mandible/metadata_mapper/format/h5.py

```diff
@@ -18,9 +18,38 @@ def parse_data(file: IO[bytes]) -> contextlib.AbstractContextManager[Any]:
 
     @staticmethod
     def eval_key(data: Any, key: Key) -> Any:
+        parsed_key, parsed_attribute = parse_key(key.key)
+        if parsed_attribute:
```

Comment:
If the input is `foo@` then `parsed_attribute == ''`. Not sure it's possible for an h5 attribute to be the empty string, but probably an edge case we should think about. If `''` is not a valid attribute name then I'd say `parse_key` should maybe raise an error in that case.
```suggestion
        if parsed_attribute is not None:
```